libs/auth/token_loader_test.go

```diff
@@ -0,0 +1,238 @@
+package auth
+
+import (
+	"context"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go/credentials/u2m"
+	"github.com/databricks/databricks-sdk-go/credentials/u2m/cache"
+	"github.com/databricks/databricks-sdk-go/httpclient/fixtures"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/oauth2"
+)
+
+type fakePersistentAuth struct {
+	token    *oauth2.Token
+	tokenErr error
+	closeErr error
+}
+
+func (f *fakePersistentAuth) Token() (*oauth2.Token, error) {
+	return f.token, f.tokenErr
+}
+
+func (f *fakePersistentAuth) Close() error {
+	return f.closeErr
+}
+
+func TestAcquireTokenDoesNotMutatePersistentAuthOpts(t *testing.T) {
+	defaultFactory := persistentAuthFactory
+	t.Cleanup(func() {
+		persistentAuthFactory = defaultFactory
+	})
+
+	opts := []u2m.PersistentAuthOption{
+		func(pa *u2m.PersistentAuth) {},
+		func(pa *u2m.PersistentAuth) {},
+	}
+	initialLen := len(opts)
+	factoryCalls := 0
+
+	persistentAuthFactory = func(ctx context.Context, providedOpts ...u2m.PersistentAuthOption) (persistentAuth, error) {
+		factoryCalls++
+		require.Len(t, providedOpts, initialLen+1)
+		require.Len(t, opts, initialLen) // original slice must not change
+		return &fakePersistentAuth{
+			token: &oauth2.Token{
+				AccessToken: "token",
+			},
+		}, nil
+	}
```

Comment:
I'd just skip that part, and use the `mockOAuthEndpointSupplier`, and then check the length of the opts, without counting the calls etc. 👍 No need for it IMO.